packages/core/src/llm/model/gateways/gateway-manager.ts

```
@@ -0,0 +1,197 @@
+import { parseModelRouterId } from '../gateway-resolver.js';
+import { PROVIDER_REGISTRY } from '../provider-registry.js';
+import { findGatewayForModel, getGatewayId, shouldEnableGateway } from './gateway-helpers.js';
+import type { GatewayAuthRequest, GatewayAuthResult, MastraModelGatewayInterface, ProviderConfig } from './base.js';
```

ESLint

[error] `./base.js` type import should occur before import of `./gateway-helpers.js`

(import/order)